OSDOCS-3435: Nutanix IPI installation (#44537)
openshift/openshift-docs: modules/installation-nutanix-installer-infra-reqs.adoc

:_content-type: CONCEPT
[id="installation-nutanix-installer-infra-reqs_{context}"]
= Environment requirements

Before you install an {product-title} cluste, review the following Nutanix AOS environment requirements.

[id="installation-nutanix-installer-infra-reqs-account_{context}"]
== Required account privileges

The installation program requires access to an account with privileges to read and create the required resources. Using an account that has global administrative privileges is the simplest way to access all of the necessary permissions.

> 

Author: Is "global administrative privileges" the correct term to describe a Prism Central admin account that has the necessary level of privileges to install a cluster. If restrictions within an organization prevent a user account from have global privileges, what subset of privileges are required.

Example of this doc: https://docs.openshift.com/container-platform/4.10/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#installation-vsphere-infrastructure_installing-vsphere-installer-provisioned

Author: @rvanderp3 To clarify - I am interpreting your comment as that the doc simply state that the account that is used to install the cluster requires administrative privileges and that the subsequent tables that detail `Roles and privileges required for installation` and `Required permissions and propagation settings` should be deleted. Is that correct?

Reviewer: @mjpytlak yes, that is correct.

Author: Done.

An additional role is required if the installation program is to create a Nutanix virtual machine folder.

.Roles and privileges required for installation
[%collapsible]
====
[cols="3a,3a,3a",options="header"]
|===
|Nutanix object for role
|When required
|Required privileges

|TBD
|TBD
|
[%hardbreaks]
`TBD`
`TBD`
`TBD`

|TBD
|TBD
|
[%hardbreaks]
`TBD`
`TBD`
`TBD`

|TBD
|TBD
|
[%hardbreaks]
`TBD`
`TBD`
`TBD`
|===
====

Additionally, the user requires some `ReadOnly` permissions, and some of the roles require permission to propogate the permissions to child objects. These settings vary depending on whether or not you install the cluster into an existing folder.

.Required permissions and propagation settings
[%collapsible]
====
[cols="3a,3a,3a,3a",options="header"]
|===
|Nutanix object
|Folder type
|Propagate to children
|Permissions required

|TBD
|TBD
|TBD
|TBD

|TBD
|TBD
|TBD
|TBD

|TBD
|TBD
|TBD
|TBD
|===
====

[id="installation-nutanix-installer-infra-reqs-limits_{context}"]
== Cluster limits

> 

Author: Cluster limits, Cluster resources, and Networking requirements are largely boilerplate. Is there anything specific to Nutanix that should be added/removed?

Author: Thank you got em all.

> :_content-type: CONCEPT

Author: Done.

[id="installation-nutanix-installer-infra-reqs_{context}"]
= Environment requirements

> Before you install an {product-title} cluste, review the following Nutanix AOS environment requirements.

Reviewer 2: Typo on "cluste,"

> Before you install an {product-title} cluste, review the following Nutanix AOS environment requirements.

Reviewer 2: Typo on "cluste,"

> Before you install an {product-title} cluste, review the following Nutanix AOS environment requirements.

Author: @vnephologist Yes - if Nutanix already has the procedure for that documented, I can reference the requirement in the Prerequisites section and link out to your content.

> Before you install an {product-title} cluste, review the following Nutanix AOS environment requirements.

Reviewer 3: @mjpytlak how about something like... 

Updating Prism Central certificates and adding them to your system trust

The installation program requires access to the Prism Central API using a valid CA-signed certificate. If you haven't already, you must replace the default self-signed certificate with a CA-signed certificate. Further details on replacing the certificate can be found in the [Nutanix AOS Security Guide](https://portal.nutanix.com/page/documents/details?targetId=Nutanix-Security-Guide-v6_1:mul-security-ssl-certificate-pc-t.html).

You must also add your Prism Central’s trusted root CA certificates to your system trust before you install an OpenShift Container Platform cluster. Using the root CA from the steps above... 

(steps similar to VMware doc in /installing/installing_vsphere/installing-vsphere-installer-provisioned.html#installation-adding-vcenter-root-certificates_installing-vsphere-installer-provisioned)

[id="installation-nutanix-installer-infra-reqs-account_{context}"]
== Required account privileges

Installing a cluster to Nutanix requires an account with administrative privileges to read and create the required resources.

[id="installation-nutanix-installer-infra-reqs-limits_{context}"]
== Cluster limits

Available resources vary between clusters. The number of possible clusters within a Nutanix environment is limited primarily by available storage space and any limitations associated with the resources that the cluster creates, and resources that you require to deploy the cluster, such a IP addresses and networks.

[id="installation-nutanix-installer-infra-reqs-resources_{context}"]
== Cluster resources

A minimum of 800 GB of storage is required to use a standard cluster.

When you deploy a {product-title} cluster that uses installer-provisioned infrastruture, the installation program must be able to create several resources in your Nutanix instance. Although these resources use 856 GB of storage, the bootstrap node is destroyed as part of the installation process.

A standard {product-title} installation creates the following resources:

> * 1 Folder

Reviewer 3: ```suggestion
```

Author: Done.

> * 1 Tag category

Reviewer 3: ```suggestion
* 1 label
```

* Virtual machines:

> ** 1 template

Reviewer 3: ```suggestion
** 1 disk image
```

Author: Done

** 1 temporary bootstrap node

> ** 3 control plan nodes

Reviewer 3: ```suggestion
** 3 control plane nodes
```

Author: Done.

** 3 compute machines

[id="installation-nutanix-installer-infra-requirements-networking_{context}"]
== Networking requirements

> You must use DHCP for the network and ensure that the DHCP server is configured to provide persistent IP addresses to the cluster machines. Additionally, create the following networking resources before you install the {product-title} cluster:

Reviewer 3: ```suggestion
You must use AHV IP Address Management (IPAM) for the network and ensure that it is configured to provide persistent IP addresses to the cluster machines. Additionally, create the following networking resources before you install the {product-title} cluster:
```

Author: Done.
